### DISCUSSION TITLE
Property not registered on control Error

### DISCUSSION THREAD

**Discussion by FraserElectronics** [AvaloniaUI/Avalonia#13168] [Q&A] [answered]
I am playing with TemplatedControl in Avalonia 11 but am having difficulty with binding properties.
I have a Value direct property in my templated control:

```
    public static readonly DirectProperty<TimePickerControl,float> ValueProperty =
    AvaloniaProperty.RegisterDirect<TimePickerControl, float>(nameof(Value), o => o.Value, ( o, v ) => o.Value = v );
    private float _value;
    public float Value
    {
        get
        {
            return _value;
        }
        set
        {
            SetAndRaise( ValueProperty, ref _value, value );
	}
    }
```

And I use the control in my main view:

```
<controls:TimePickerControl Caption="Choose Value" Value="{Binding TimeValue,Mode=TwoWay}"
```

However, when I run my application, I get the run-time error:

```
System.ArgumentException: 'Property 'Value not registered on 'MyApp.Controls.TimePickerControl'
```

What am I missing ?

Andy

**Comment by rabbitism** [comment]
Can you prepare a minimal sample? I tried in a new project, and it works well.

**Reply by FraserElectronics** [reply]
Unfortunately, if I make a sample it works fine :-(
It was working, the only thing I did was rename Value from UserValue as it seemed a more consistent name and that is when I started having this problem !

Andy

**Reply by timunie** [reply]
Have you remembered to also rename property registration? Its important to match the pattern.

**Reply by FraserElectronics** [reply]
Did I not show the registration code above or is there somewhere else I have forgotten to update?
Is this the registration code?

```
AvaloniaProperty.RegisterDirect<TimePickerControl, float>(nameof(Value), o => o.Value, ( o, v ) => o.Value = v );
```

Andy

**Reply by FraserElectronics** [reply]
What is annoying is that Intellisense shows Value so it sees that as a property on my control :-(

**Reply by FraserElectronics** [reply, marked as answer]
Never mind, I was being an idiot - I was passing the wrong control type when registering the property despite what my code shows above.
Thanks for your help 👍

Andy
